phoenix-core/src/main/java/org/apache/phoenix/compile/GroupByCompiler.java

```diff
@@ -63,9 +63,10 @@
         private final boolean isOrderPreserving;
```

Comment:
Do you need this boolean any more? Seems like we always should use ```orderPreservingTrackInfos``` to determine if its order preseving right?

Reply:
Yes , I think the member variable `isOrderPreserving` is required because it saved the result of `OrderPreservingTracker.isOrderPreserving()`.  `GroupBy.isOrderPreserving()` is used by many classes under compile and execute package.
FYI, the `GroupBy` class is a immutable class , when we compile the `group by clause` at difference time(we compile it twice in `QueryCompiler.compileSingleFlatQuery`) , a new GroupBy instance  is created.